#! /bin/sh /usr/share/dpatch/dpatch-run

## 18_vdr-1.3.38-root-fix.dpatch by Thomas Schmidt <tschmidt@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Do not try to change capabilities when running as root

@DPATCH@
diff -urNad vdr-1.3.38/vdr.c /tmp/dpep.tm8hXN/vdr-1.3.38/vdr.c
--- vdr-1.3.38/vdr.c	2006-01-08 12:49:03.000000000 +0100
+++ /tmp/dpep.tm8hXN/vdr-1.3.38/vdr.c	2006-01-09 20:12:03.632027750 +0100
@@ -339,14 +339,16 @@
 
   if (getuid() == 0) {
      StartedAsRoot = true;
-     if (!SetKeepCaps(true))
-        return 2;
-     if (!SetUser(VdrUser))
-        return 2;
-     if (!SetKeepCaps(false))
-        return 2;
-     if (!SetCapSysTime())
-        return 2;
+     if (strcmp(VdrUser, "root")) {
+        if (!SetKeepCaps(true))
+           return 2;
+        if (!SetUser(VdrUser))
+           return 2;
+        if (!SetKeepCaps(false))
+           return 2;
+        if (!SetCapSysTime())
+           return 2;
+        }
      }
 
   // Help and version info:
